File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/status/retrofit/StatusRequestBodyConverter.java
package org.dvbviewer.controller.data.status.retrofit;

import org.dvbviewer.controller.entities.Status;

import okhttp3.MediaType;
import okhttp3.RequestBody;
import retrofit2.Converter;

final class StatusRequestBodyConverter implements Converter<Status, RequestBody> {

    private static final MediaType MEDIA_TYPE = MediaType.parse("text");

    @Override public RequestBody convert(Status value) {
        return RequestBody.create(MEDIA_TYPE, value.toString());
    }

}

File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/status/retrofit/StatusRequestBodyConverter.kt
package org.dvbviewer.controller.data.status.retrofit

import okhttp3.MediaType
import okhttp3.RequestBody
import org.dvbviewer.controller.entities.Status
import retrofit2.Converter

internal class StatusRequestBodyConverter : Converter<Status, RequestBody> {

    override fun convert(value: Status): RequestBody {
        return RequestBody.create(MEDIA_TYPE, value.toString())
    }

    companion object {

        private val MEDIA_TYPE = MediaType.parse("text")
    }

}